src/fileformat/types/resource_table/resource_table.cpp

```diff
@@ -213,6 +241,76 @@ ResourceTable::resourcesIterator ResourceTable::end() const
 	return table.end();
 }
 
+/**
+ * Compute icon hashes - CRC32, MD5, SHA256.
+ * @param rOwner Pointer to input file
+ */
+void ResourceTable::computeIconHashes(const FileFormat *rOwner)
+{
+	std::vector<Resource *> iconsWithName;
+	std::vector<Resource *> iconsWithId;
+	std::vector<std::uint8_t> iconHashBytes;
+
+	for(auto& resource : table)
+	{
+		if(!resource.isUsedForIconhash() || resource.getType() != "Icon")
```

Comment:
Is it necessary to use strings to distinguish what type of resource it is? Would it be possible to solve it by comparing some enum values instead of strings?